nucliadb/nucliadb/writer/api/v1/upload.py

```diff
@@ -294,7 +294,8 @@ async def _tus_post(
     await storage_manager.start(dm, path=path, kbid=kbid)
     await dm.save()
 
-    location = f"{request['path']}/{upload_id}"
+    api.url_path_for
+    location = api.url_path_for("Upload information", kbid=kbid, upload_id=upload_id)
```

Comment:
Starlette 0.35 (required by fastapi 0.109, required to fix dependabot warning) changes the behaviour of `request['path']`, the old behaviour was a bug. https://github.com/encode/starlette/pull/2400

Changing it to use a higher-level API which works with both versions and should always be correct.

As a note, our use of the `Location` header here is questionable, since we return something like `/kb/...` but we expect the client to send a request to `/api/v1/kb/...` which is non-standard. Of course, it works well with our clients, but still, it's a bit unexpected for people not using the SDK.

Reply:
careful though, there are several endpoints with the "upload information" name

Reply:
Of course, and that's on purpose :smile:

`url_path_for` matches on name and parameters. So this will match the correct endpoint of the three (it gets passed `**request.path_params` although that's not showing here :/.